pilot/pkg/serviceregistry/external/controller_test.go

```diff
@@ -97,19 +83,19 @@ func TestController(t *testing.T) {
 			Resolution: networking.ServiceEntry_STATIC,
 		},
 	}
-	expectedCount := 7 // 1 service + 6 instances
+	expectedCount := int64(7) // 1 service + 6 instances
 
 	_, err = configController.Create(cfg)
 	if err != nil {
 		t.Fatalf("error occurred crearting ServiceEntry config: %v", err)
 	}
 
 	if err := retry.UntilSuccess(func() error {
-		if c := getCountAndReset(); c != expectedCount {
-			return fmt.Errorf("got %d notifications from controller, want %d", c, 7)
+		if gotcount := atomic.AddInt64(&count, 0); gotcount != expectedCount {
```

Comment:
Before we set this to 0, now we just add 0? Should we set this to 0 here?

/hold

Reply:
setting it to `0` was the problem :-)

Reply:
We have the retry though. So the test expects 7. What if the first time it gives 3, then the second time it gives 4. Now the test passes?

Or if it fails once and got 1, then succeeds and gets 7 more, now it will be at 8 and never pass?

I may be misunderstanding it though

Reply:
@howardjohn I think the new code is correct. The retry loop is really just getting the current value (it should probably be replaced with `atomic.LoadInt64` for clarity). There should only be 7 events total and we're waiting up to the timeout for all 7 to occur.

Or am I missing something?

Reply:
Ok yeah I think you are right. I was thinking we reran the whole test. Thanks
/hold-cancel